cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -678,16 +746,18 @@ private Location createTempLocation(String fileName) throws IOException {
    * Creates a {@link V1PodSpec} for specifying pod information for running the given runnable.
    *
    * @param runtimeConfigLocation the {@link Location} containing the runtime config archive
-   * @param runtimeSpec the specifiction for the {@link TwillRunnable} and its resources requirements
+   * @param runtimeSpecs the specifiction for the {@link TwillRunnable} and its resources requirements
    * @return a {@link V1PodSpec}
    */
   private V1PodSpec createPodSpec(Location runtimeConfigLocation,
-                                  RuntimeSpecification runtimeSpec, V1VolumeMount... extraMounts) {
-    String runnableName = runtimeSpec.getName();
+                                  Map<String, RuntimeSpecification> runtimeSpecs, V1VolumeMount... extraMounts) {
     String workDir = "/workDir-" + twillRunId.getId();
 
     V1Volume podInfoVolume = createPodInfoVolume(podInfo);
-    V1ResourceRequirements resourceRequirements = createResourceRequirements(runtimeSpec.getResourceSpecification());
+
+    RuntimeSpecification mainRuntimeSpec = getMainRuntimeSpecification(runtimeSpecs);
+    V1ResourceRequirements initContainerResourceRequirements =
+      createResourceRequirements(mainRuntimeSpec.getResourceSpecification());
```

Comment:
The resource requirements should be a sum of all the runnable requirements so that the pod can accommodate all the containers.

Reply:
resource requirements are specified per container. I thought kube will aggregate resource requirements of all containers in a pod to come up with resource requirement of the pod.